docs/cms-query-integration.md

```diff
@@ -0,0 +1,64 @@
+# CMS Query Integration
+
+The Search module overrides the default search functionality to query the specialized search index in place of a standard MySQL query. This means all default search operations using the CMS search APIs for posts, users and terms such as `WP_Query`, `WP_User_Query`, `WP_Term_Query` and the REST API search endpoints will transparently make use of the search index by default.
+
+It is possible to control this behaviour and thus use the search index for non search queries that might otherwise be too slow to do in the database such as meta queries.
+
+Each query class accepts an array of parameters. ElasticPress checks for the value of an `ep_integrate` parameter, if this is true the query is performed by Elasticsearch instead of MySQL. For example:
+
+```php
+$posts = new WP_Query( [
+	'ep_integrate' => true, // Use Elasticsearch.
+	'post_type' => 'project',
+	'meta_query' => [
+		[
+			'key' => 'color',
+			'value' => [ 'blue', 'green' ],
+			'compare' => 'IN'
+		],
+	],
+] );
+```
+
+Note in the above example you would need a PHP Code Sniffer exception for using the meta query to pass [Automated Code Review](docs://guides/code-review/README.md).
+
+You can also circumvent the use of Elasticsearch for searches by explicitly setting `ep_integrate` to false in the query arguments or by using the `pre_get_posts` action and checking for certain conditions:
+
+```php
+add_action( 'pre_get_posts', function ( WP_Query $query ) {
+	// Avoid using Elasticsearch for the 'project' post type.
+	if ( ! $query->is_search() || $query->get( 'post_type' ) !== 'project' ) {
+		return;
+	}
+
+	$query->set( 'ep_integrate', false );
+}, 20 );
+```
+
+## Autosuggest
+Because of the way Elasticsearch works it performs better with complete search terms as you might expect to be submitted from a search form.
+
+In some cases you might want to provide a dynamic interface where results are fetched as the user types. The standard method of search will not produce good results until a full word has been typed out so Altis provides a means of analysing and fetching results based on partial search terms.
+
+In the same way `ep_integrate` can be passed to any query class you can pass the parameter `autosuggest`:
+
+```php
+$posts = new WP_Query( [
+	'autosuggest' => true,
```

Comment:
Is `ep_integrate` required too? We should state explicitly either way.

Reply:
Ah yeah, I'll make it clearer that `ep_integrate` defaults to true when the `s` parameter is used.